elide-core/src/main/java/com/yahoo/elide/graphql/PersistentResourceFetcher.java

```diff
@@ -79,6 +119,117 @@ private void logContext(RelationshipOp operation, Environment environment) {
+        }
+    }
+
+    /**
+     * FETCH top level entity either by ids or as a collection
+     * @param requestScope request scope
+     * @param recordType record type
+     * @param ids list of ids (can be NULL)
+     * @param sort sort by ASC/DESC
+     * @param offset pagination offset argument
+     * @param first pagination first argument
+     * @param filters filter params
+     * @return list of persistent resource objects
+     */
+    private Object fetchObject(RequestScope requestScope, Class recordType, Optional<List<String>> ids,
+                               Optional<String> sort, Optional<String> offset,
+                               Optional<String> first, Optional<String> filters) {
+        /* fetching a collection */
+        if(!ids.isPresent()) {
+            List<PersistentResource> records = new ArrayList<>(PersistentResource.
+                    loadRecords(recordType, requestScope));
+
+            //TODO: paginate/filter/sort
+            return records;
+
+        } else { /* fetching by id */
+            List<String> idList = ids.get();
+
+            /* handle empty list of ids */
+            if(idList.isEmpty()) {
+                throw new IllegalArgumentException("Empty list passed to ids");
+            }
+
+            /* access records from internal db and return */
+            HashSet recordSet = new HashSet();
+            for (String id : idList) {
+                recordSet.add(PersistentResource.loadRecord(recordType, id, requestScope));
+            }
+            return recordSet;
+        }
+    }
+
+    /**
+     * FETCH attributes of top level entity
+     * @param requestScope request scope
+     * @param parentResource parent object
+     * @param field Field type
+     * @param outputType GraphQL object output type
+     * @param ids list of ids
+     * @param sort sort by ASC/DESC
+     * @param offset pagination offset argument
+     * @param first pagination first argument
+     * @param filters filter params
+     * @return list of persistent resource objects
+     */
+    private Object fetchObject(RequestScope requestScope, PersistentResource parentResource, Field field,
+                               GraphQLType outputType, Optional<List<String>> ids,
+                               Optional<String> sort, Optional<String> offset,
+                               Optional<String> first, Optional<String> filters) {
+
+        //TODO: Do we need P/F/S? If so, do it!
+
+        if(outputType instanceof GraphQLObjectType) {
+            DataStoreTransaction tx = requestScope.getTransaction();
+            Object obj =
+                    tx.getRelation(tx, parentResource.getObject(), field.getName(),
+                            null, null, null, requestScope);
+            return new PersistentResource(obj, parentResource, requestScope.getUUIDFor(obj), requestScope);
+        }
+
+        else if(outputType instanceof GraphQLScalarType || outputType instanceof GraphQLEnumType) {
```

Comment:
Instead of checking the types, you should just check to see if this is an attribute (see above comment).  Regardless of type, we should just call parentResource.getAttribute(if this is an attribute)